### PR TITLE
WT-10528 test_wt8246_compact_rts_data_correctness retry to insert the data during WT_ROLLBACK

### PR DESCRIPTION
**Pull request by raviprakashgiri29**
Retry to insert the data till we get WT_ROLLBACK.

### CODE REVIEW

File: test/csuite/wt8246_compact_rts_data_correctness/main.c [hunk 1]
```diff
@@ -337,12 +337,11 @@ large_updates(WT_SESSION *session, const char *uri, char *value, int commit_ts)
         cursor->set_key(cursor, i + 1);
         val = (uint64_t)__wt_random(&rnd);
         cursor->set_value(cursor, val, val, val, value);
-        if ((ret = cursor->insert(cursor)) == WT_ROLLBACK)
-            testutil_check(session->rollback_transaction(session, NULL));
-        else {
-            testutil_check(ret);
-            testutil_check(session->commit_transaction(session, tscfg));
-        }
+        while ((ret = cursor->insert(cursor)) == WT_ROLLBACK)
```
Comment (quchenhao) [inline test/csuite/wt8246_compact_rts_data_correctness/main.c:340]: Should we have a limit on how many times we retry?

Comment (raviprakashgiri29) [reply test/csuite/wt8246_compact_rts_data_correctness/main.c]: Fixed in 2f1d787cbcf8a28614f0267282e768cf338d93a4 to retry for 5 times and then abort it it still returns WT_ROLLBACK.

File: test/csuite/wt8246_compact_rts_data_correctness/main.c [hunk 2]
```diff
@@ -337,12 +339,14 @@ large_updates(WT_SESSION *session, const char *uri, char *value, int commit_ts)
         cursor->set_key(cursor, i + 1);
         val = (uint64_t)__wt_random(&rnd);
         cursor->set_value(cursor, val, val, val, value);
-        if ((ret = cursor->insert(cursor)) == WT_ROLLBACK)
```
Comment (quchenhao) [inline test/csuite/wt8246_compact_rts_data_correctness/main.c:340]: I don't think this is correct. We need to rollback the transaction first and then retry.

Comment (raviprakashgiri29) [reply test/csuite/wt8246_compact_rts_data_correctness/main.c:340]: Fixed 457b8b94a160b3e70f626937ccbd32dd1c47e471

File: test/csuite/wt8246_compact_rts_data_correctness/main.c [hunk 3]
```diff
@@ -337,12 +339,16 @@ large_updates(WT_SESSION *session, const char *uri, char *value, int commit_ts)
...
+        while (((ret = cursor->insert(cursor)) == WT_ROLLBACK) && retry_attempts < MAX_RETRIES) {
```
Comment (quchenhao) [inline test/csuite/wt8246_compact_rts_data_correctness/main.c:342]: You need to call begine_transaction before you insert again?

**Pull request merged**